submariner/multiple-cable-drivers.md

```diff
@@ -0,0 +1,415 @@
+    location: cloud
+    env: production
+  connectionPolicy: no-connect
+  priority: 3
+```
+
+- We only want certain clusters to access a database server in another cluster and no other clusters to connect to one another. We
+label the server and client `Clusters` appropriately and define a single policy with corresponding label selectors. Since there's
+no other policy to match client cluster pairs, they will not be connected.
+
+```yaml
+apiVersion: v1
+kind: ClusterConnectionPolicies
+metadata:
+  name: client-server-policy
+spec:
+  leftClusterSelector:
+    database-role: server
+  rightClusterSelector:
+    database-role: client
+  topology: client-server
+  connectionPolicy: connect
+  cableDriver:
+      name:  "vxlan"
+   priority: 3
+```
+
+- We want any cluster to access a database server in another cluster but no other clusters to connect. We label the server
+`Cluster` appropriately and define a single policy with the corresponding label selector and an empty selector to match all.
+
+```yaml
+apiVersion: v1
+kind: ClusterConnectionPolicies
+metadata:
+  name: client-server-policy
+spec:
+  leftClusterSelector:
+    database-server: true
+  rightClusterSelector:
+  topology: client-server
+  connectionPolicy: connect
+  cableDriver:
+      name:  "vxlan"
+   priority: 3
+```
+
+### Multiple Cable driver configuration
+
+The cable driver configuration in Submariner is relatively limited, as one can only select one driver to use at a time and there’s minimal
+support for customization of the configuration. The goals of this proposal for the extension of the cable driver configuration are:
+
+- Maintain backward compatibility with the current cable driver configuration
+- Create a custom cable driver configuration that allows for the setup of one, many or combined drivers. In addition, the configuration
+should allow for further customizations based on the selected driver(s).
+
+Cable driver configuration parameters fall into two categories based on their scope:
+
+- Gateway Pod Local - these are parameters that hold configuration parameters specific to each Pod (values vary from Pod to Pod).
+- Namespace Local - these are configuration parameters that you would like to share across a namespace (in this case the submariner-operator
+namespace).
+
+The proposed approach for additional submariner cable driver configuration parameters is:
+
+- To continue to use the existing environment variables injected into the Gateway Pod (with the exception of PSKs/secrets). These environment
+variables will come from the `submariner.io_submariners.yaml`.
+- To extend `submariner.io_submariners.yaml` with some new global parameters for cable driver configuration in a cluster set.
+- To use a volume mounted ConfigMap for more detailed cable driver configurations (as these may vary).
+- To use mounted secrets for parameters that need to be protected, e.g. the Pre-Shared Key (PSK) and to abandon the use of environment variables
+for secrets.
```

Comment:
This seems like a plethora of ideas that are pre-cursors or perhaps tangential to this EP, I suggest to tackle them separately instead of lumping them in here.

This EP should focus on the policy aspects not on extending or refactoring cable driver customization options.

Reply:
Fair enough